packages/atomic/src/components/common/item-link/item-link.spec.ts

```diff
@@ -0,0 +1,432 @@
+import {renderFunctionFixture} from '@/vitest-utils/testing-helpers/fixture';
+import '@vitest/browser/matchers.d.ts';
+import {html} from 'lit';
+import {describe, it, vi, expect, beforeEach} from 'vitest';
+import {
+  renderLinkWithItemAnalytics,
+  bindAnalyticsToLink,
+  type ItemLinkProps,
+} from './item-link';
+
+// Mock external dependencies
+vi.mock('@/src/utils/xss-utils', () => ({
+  filterProtocol: vi.fn((url: string) => url),
+}));
+
+describe('#renderLinkWithItemAnalytics', () => {
+  const renderComponent = async (
+    props: Partial<ItemLinkProps> = {},
+    children = html`<span>Default Link Text</span>`
+  ) => {
+    const defaultProps: ItemLinkProps = {
+      href: 'https://example.com',
+      onSelect: vi.fn(),
+      onBeginDelayedSelect: vi.fn(),
+      onCancelPendingSelect: vi.fn(),
+      ...props,
+    };
+
+    return await renderFunctionFixture(
+      html`${renderLinkWithItemAnalytics({props: defaultProps})(children)}`
+    );
+  };
+
+  beforeEach(() => {
+    vi.clearAllMocks();
+  });
+
+  describe('basic rendering', () => {
+    it('should render link element with required props', async () => {
+      const element = await renderComponent({
+        href: 'https://test.com',
+      });
+
+      const link = element.querySelector('a');
+      expect(link).toBeInTheDocument();
+      expect(link).toHaveAttribute('href', 'https://test.com');
+      expect(link).toHaveAttribute('target', '_self');
+    });
+
+    it('should render children correctly', async () => {
+      const element = await renderComponent(
+        {},
+        html`<span class="test-child">Custom Link Text</span>`
+      );
+
+      const child = element.querySelector('.test-child');
+      expect(child).toHaveTextContent('Custom Link Text');
+    });
+
+    it('should render multiple children correctly', async () => {
+      const element = await renderComponent(
+        {},
+        html`
+          <span class="child-1">Part 1</span>
+          <span class="child-2">Part 2</span>
+        `
+      );
+
+      expect(element.querySelector('.child-1')).toHaveTextContent('Part 1');
+      expect(element.querySelector('.child-2')).toHaveTextContent('Part 2');
+    });
+
+    it('should handle empty children', async () => {
+      const element = await renderComponent({}, html``);
+
+      const link = element.querySelector('a');
+      expect(link).toBeInTheDocument();
+      expect(link).toHaveTextContent('');
+    });
+  });
+
+  describe('prop handling', () => {
+    it('should apply className when provided', async () => {
```

Comment:
We should also test that it has not default class, i.e., that when you don't provide a className in the props, the rendered link will have no class.

The way you end up testing this will depend on whether you apply this comment https://github.com/coveo/ui-kit/pull/5607/files#r2195520452. If you do, it's gonna be similar to the tests you have for the tabIndex.